mina.netty/src/main/java/org/jboss/netty/channel/socket/nio/NioWorker.java

```diff
@@ -100,6 +108,9 @@ protected final long getMaximumProcessTaskQueueTimeNanos() {
 
     @Override
     protected boolean read(SelectionKey k) {
+        if (k.channel() instanceof DatagramChannel) {
+            return readUdp(k);
+        }
```

Comment:
As we talked I will try to abstract it into some interface and move the read impl it to NioSocketChannel and NioDatagramChannel. Not sure if it helps with the readability as the worker logic is moved somewhere else. I will try that in separate PR.